api/staleness.py

```diff
@@ -49,26 +56,54 @@ def _validate_input_data(body):
         abort(HTTPStatus.BAD_REQUEST, f"Validation Error: {str(e.messages)}")
 
 
-def _update_hosts_staleness_async(org_id, app):
+def _update_hosts_staleness_async(identity, app, staleness):
     with app.app_context():
+        threadctx.request_id = None
         logger.debug("Starting host staleness update thread")
         try:
-            logger.info(f"Querying hosts for org_id: {org_id}")
-            hosts_query = Host.query.filter(Host.org_id == org_id)
+            logger.info(f"Querying hosts for org_id: {identity.org_id}")
+            hosts_query = Host.query.filter(Host.org_id == identity.org_id)
             num_hosts = hosts_query.count()
+            st = staleness_timestamps()
+            staleness = serialize_staleness_to_dict(staleness)
+            list_of_events_params = []
             if num_hosts > 0:
-                logger.info(f"Found {num_hosts} hosts for org_id: {org_id}")
+                logger.info(f"Found {num_hosts} hosts for org_id: {identity.org_id}")
                 for host in hosts_query.yield_per(500):
                     host._update_all_per_reporter_staleness()
+                    serialized_host = serialize_host(host, for_mq=False, staleness_timestamps=st, staleness=staleness)
+
+                    # Create host update event and append it to an array
+                    event, headers = _build_host_updated_event_params(serialized_host, host, identity)
+                    list_of_events_params.append((event, headers, str(host.id)))
                 hosts_query.session.commit()
 
-                delete_cached_system_keys(org_id=org_id, spawn=True)
+                # After a successful commit to the db
+                # call all the events in the list
+                for event, headers, host_id in list_of_events_params:
+                    app.event_producer.write_event(event, host_id, headers, wait=True)
+
+                delete_cached_system_keys(org_id=identity.org_id, spawn=True)
             logger.info("Leaving host staleness update thread")
```

Comment:
I know this wasn't done in this PR, but I just realized that the "Starting" log is debug-level, and this one should be as well
```suggestion
            logger.debug("Leaving host staleness update thread")
```